python_modules/dagster/dagster/core/execution/plan/compute_generator.py

```diff
@@ -109,8 +109,15 @@ def _validate_and_coerce_solid_result_to_iterator(result, context, output_defs):
             )
 
         for output_def, element in zip(output_defs, result):
-            metadata = context.get_output_metadata(output_def.name)
-            yield Output(output_name=output_def.name, value=element, metadata=metadata)
+            if isinstance(element, Output):
+                yield Output(
+                    output_name=output_def.name,
+                    value=element.value,
+                    metadata_entries=element.metadata_entries,
+                )
```

Comment:
hmm so we always overwrite `output_name` here? I feel like that may be confusing if someone explicitly sets the output names. We should add test cases for that [1]